src/v/config/configuration.h

```diff
@@ -170,6 +170,7 @@ struct configuration final : public config_store {
     property<int16_t> id_allocator_batch_size;
     property<bool> enable_sasl;
     property<std::optional<bool>> kafka_enable_authorization;
+    property<std::optional<ss::sstring>> kafka_mtls_principal_mapping_rules;
```

Comment:
I was under the impression that this was going to be a list of rules so that the user didn't have to encode multiple rules in a single string.